pkg/validation-webhook/validation-webhook.go

```diff
@@ -0,0 +1,254 @@
+package main
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"flag"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+
+	vckv1alpha1 "github.com/IntelAI/vck/pkg/apis/vck/v1alpha1"
+	"github.com/golang/glog"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+)
+
+//Config struct
+type Config struct {
+	CertFile string
+	KeyFile  string
+}
+
+var scheme = runtime.NewScheme()
+
+var codecs = serializer.NewCodecFactory(scheme)
+
+type admitFunc func(v1beta1.AdmissionReview) *v1beta1.AdmissionResponse
+
+func toAdmissionResponse(err error) *v1beta1.AdmissionResponse {
+	return &v1beta1.AdmissionResponse{
+		Result: &metav1.Status{
+			Message: err.Error(),
+		},
+	}
+}
+
+func admitVolumeManager(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	glog.V(2).Info("admitting volume manager")
+
+	raw := ar.Request.Object.Raw
+	vm := vckv1alpha1.VolumeManager{}
+	deserializer := codecs.UniversalDeserializer()
+	if _, _, err := deserializer.Decode(raw, nil, &vm); err != nil {
+		glog.Error(err)
+		return toAdmissionResponse(err)
+	}
+
+	reviewResponse := validateVolumeManager(vm)
+
+	return reviewResponse
+}
+
+func validateNFS(vc vckv1alpha1.VolumeConfig) string {
+	errs := []string{}
+	if len(vc.Labels) == 0 {
+		errs = append(errs, "labels cannot be empty.")
+	}
+
+	if _, ok := vc.Options["server"]; !ok {
+		errs = append(errs, "server has to be set in options.")
+	}
+
+	if _, ok := vc.Options["path"]; !ok {
+		errs = append(errs, "path has to be set in options.")
+	}
+
+	if vc.AccessMode == "" {
+		errs = append(errs, "accessMode has to be set.")
+	}
+
+	return strings.Join(errs, " ")
+}
+
+func validateS3(vc vckv1alpha1.VolumeConfig) string {
+	errs := []string{}
+	if len(vc.Labels) == 0 {
+		errs = append(errs, "labels cannot be empty.")
+	}
+
+	if vc.Replicas < 1 {
+		errs = append(errs, "replicas cannot be empty or less than 1.")
+	}
+
+	if _, ok := vc.Options["awsCredentialsSecretName"]; !ok {
+		errs = append(errs, "awsCredentialsSecretName key has to be set in options.")
+	}
+
+	if _, ok := vc.Options["sourceURL"]; !ok {
+		errs = append(errs, "sourceURL has to be set in options.")
+	}
+
+	if _, err := url.ParseRequestURI(vc.Options["sourceURL"]); err != nil {
+		errs = append(errs, "sourceURL has to be a valid URL.")
+	}
+
+	if sourceURL, ok := vc.Options["endpointURL"]; ok {
```

Comment:
I guess the other stuff can also be included here, like the timeoutForDataDownload, dataPath?